client/app/pages/dashboards/components/DashboardListEmptyState.tsx

```diff
@@ -1,10 +1,16 @@
-import React from "react";
-import PropTypes from "prop-types";
+import * as React from "react";
+import * as PropTypes from "prop-types";
 import BigMessage from "@/components/BigMessage";
 import NoTaggedObjectsFound from "@/components/NoTaggedObjectsFound";
 import EmptyState from "@/components/empty-state/EmptyState";
 
-export default function DashboardListEmptyState({ page, searchTerm, selectedTags }) {
+export interface DashboardListEmptyStateProps {
```

Comment:
I tried to include https://github.com/milesj/babel-plugin-typescript-to-proptypes. This would have required a switch from ts-loader to handling the typescript transpiling in babel (@babel/preset-typescript). 
While trying this out babel uncovered a few inconsistencies that would need to be fixed first (e.g. version.json not containing a valid JSON).

Without the plugin there are still a few options for proptypes:
* All new code does not really require proptypes. We will see TypeScript compiler errors instead.
* Keeping both for essential components that are primarily called from JS code (like here)
* Existing files could also use the following code to infer types from existing proptypes:

```
export type DashboardListEmptyStateProps = InferProps<typeof DashboardListEmptyState.propTypes>

export function DashboardListEmptyState({ page, searchTerm, selectedTags }: DashboardListEmptyStateProps) {
```